src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -2717,13 +2717,13 @@ private bool CheckInArgTypes(TexlNode left, TexlNode right)
                         }
 
                         var typedName = names.Single();
-                        if (!typeLeft.CoercesTo(typedName.Type))
```

Comment:
Why these changes? Can you leave a comment about the logic?

Reply:
updated it and added comments.